src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/ConfiguredCancelableAsyncEnumerableTests.netcoreapp.cs

```diff
@@ -0,0 +1,216 @@
+            CancellationToken token1 = new CancellationTokenSource().Token;
+            CancellationToken token2 = new CancellationTokenSource().Token;
+
+            // No WithCancellation call
+            enumerable = new TrackFlagsAsyncEnumerable();
+            enumerable.GetAsyncEnumerator();
+            Assert.Equal(CancellationToken.None, enumerable.CancellationToken);
+
+            // Only ConfigureAwait call
+            enumerable = new TrackFlagsAsyncEnumerable();
+            enumerable.ConfigureAwait(false).GetAsyncEnumerator();
+            Assert.Equal(CancellationToken.None, enumerable.CancellationToken);
+
+            // Single WithCancellation call
+            enumerable = new TrackFlagsAsyncEnumerable();
+            enumerable.WithCancellation(token1).GetAsyncEnumerator();
+            Assert.Equal(token1, enumerable.CancellationToken);
+
+            // Unnecessary multiple calls; only last one is used
+            enumerable = new TrackFlagsAsyncEnumerable();
+            enumerable.WithCancellation(token1).WithCancellation(token2).GetAsyncEnumerator();
+            Assert.Equal(token2, enumerable.CancellationToken);
+
+            // Before ConfigureAwait
+            enumerable = new TrackFlagsAsyncEnumerable();
+            enumerable.WithCancellation(token1).ConfigureAwait(false).GetAsyncEnumerator();
+            Assert.Equal(token1, enumerable.CancellationToken);
+
+            // After ConfigureAwait
+            enumerable = new TrackFlagsAsyncEnumerable();
+            enumerable.ConfigureAwait(false).WithCancellation(token1).GetAsyncEnumerator();
+            Assert.Equal(token1, enumerable.CancellationToken);
+        }
+
+        private sealed class TrackFlagsAsyncEnumerable : IAsyncEnumerable<int>, IAsyncEnumerator<int>, IValueTaskSource<bool>, IValueTaskSource
+        {
+            public ValueTaskSourceOnCompletedFlags Flags;
+            public CancellationToken CancellationToken;
+
+            public IAsyncEnumerator<int> GetAsyncEnumerator(CancellationToken cancellationToken = default)
+            {
+                CancellationToken = cancellationToken;
+                return this;
+            }
+
+            public ValueTask<bool> MoveNextAsync() => new ValueTask<bool>(this, 0);
+            public int Current => throw new NotImplementedException();
+            public ValueTask DisposeAsync() => new ValueTask(this, 0);
+
+            public void OnCompleted(Action<object> continuation, object state, short token, ValueTaskSourceOnCompletedFlags flags) => Flags = flags;
+            public ValueTaskSourceStatus GetStatus(short token) => ValueTaskSourceStatus.Pending;
+            public bool GetResult(short token) => throw new NotImplementedException();
+            void IValueTaskSource.GetResult(short token) => throw new NotImplementedException();
+        }
+
+        private sealed class EnumerableWithDelayToAsyncEnumerable<T> : IAsyncEnumerable<T>, IAsyncEnumerator<T>
```

Comment:
>EnumerableWithDelayToAsyncEnumerable [](start = 29, length = 36)

This reminded me of a question: are we going to offer an API to convert an IEnumerable into an IAsyncEnumerable? 
For example, in your ZipAsync scenarios, but one of the enumerables is synchronous and the other is async.

Reply:
Not planning to.  This was discussed and the thinking is that such a thing is mainly only valuable if you're planning to use LINQ APIs, so it should live where a LINQ implementation lives, and Ix's async support already has such a conversion method.